### PR TITLE
plan: only plans that have apply will add cache.

### PR DESCRIPTION
**Pull request by hanfei1991**
only plans that have apply will add cache.
@shenli @coocood @zimulala PTAL

### CODE REVIEW

**Comment by shenli** [comment]
LGTM

File: plan/optimizer.go [hunk 1]
```diff
@@ -102,14 +102,16 @@ func logicalOptimize(flag uint64, logic LogicalPlan, ctx context.Context, alloc
...
-func physicalOptimize(logic LogicalPlan, allocator *idAllocator) (PhysicalPlan, error) {
+func physicalOptimize(flag uint64, logic LogicalPlan, allocator *idAllocator) (PhysicalPlan, error) {
 	info, err := logic.convert2PhysicalPlan(&requiredProperty{})
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
 	pp := info.p
 	pp = EliminateProjection(pp)
-	addCachePlan(pp, allocator)
+	if flag&(flagDecorrelate) > 0 {
```
Comment (ngaut) [inline plan/optimizer.go]: The source code looks like it's not formatted .

File: plan/optimizer.go [hunk 2]
```diff
@@ -102,14 +102,16 @@ func logicalOptimize(flag uint64, logic LogicalPlan, ctx context.Context, alloc
...
+		addCachePlan(pp, allocator)
+	}
 	log.Debugf("[PLAN] %s", ToString(pp))
```
Comment (ngaut) [inline plan/optimizer.go]: Remove the debug message

File: plan/optimizer.go (hunk 1 above)
Comment (hanfei1991) [inline plan/optimizer.go]: `go fmt` force it.

**Comment by shenli** [comment]
@ngaut PTAL

**Comment by ngaut** [comment]
LGTM

**Pull request merged**